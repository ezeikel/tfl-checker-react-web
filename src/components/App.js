import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import styled from "styled-components";
import { library } from "@fortawesome/fontawesome-svg-core";
import { fab } from "@fortawesome/free-brands-svg-icons";
import {
  faWalking,
  faBicycle,
  faBus,
  faTrain,
  faSubway,
  faTram,
  faShip,
  faExclamationTriangle,
  faRoute,
  faSpinnerThird,
  faChevronRight,
  faCircle,
  faBusAlt,
  faArrowRight,
  faArrowLeft,
} from "@fortawesome/pro-duotone-svg-icons";
import { faMapMarkerAlt } from "@fortawesome/pro-solid-svg-icons";
import store from "../redux/store";
import GlobalStyle from "../GlobalStyle";
import Header from "./Header";
import Footer from "./Footer";
import Main from "./Main";

library.add(
  fab,
  faWalking,
  faBicycle,
  faBus,
  faBusAlt,
  faTrain,
  faSubway,
  faTram,
  faShip,
  faMapMarkerAlt,
  faExclamationTriangle,
  faRoute,
  faSpinnerThird,
  faChevronRight,
  faCircle,
  faArrowRight,
  faArrowLeft,
);

const Container = styled.div`
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: auto 1fr;
  background-color: var(--color-background);
  min-height: 100vh;
`;

const App = () => (
  <Provider store={store}>
    <GlobalStyle />
    <Container>
      <Router>
        <Header />
        <Main />
        <Footer />
      </Router>
    </Container>
  </Provider>
);

export default App;
