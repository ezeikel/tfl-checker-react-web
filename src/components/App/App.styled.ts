/* eslint-disable import/prefer-default-export */

import styled from "styled-components";

export const Container = styled.div`
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: auto 1fr;
  background-color: var(--color-background);
  min-height: 100vh;
`;
