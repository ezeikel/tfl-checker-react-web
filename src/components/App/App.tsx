import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { library } from "@fortawesome/fontawesome-svg-core";
import { fab } from "@fortawesome/free-brands-svg-icons";
import {
  faWalking,
  faBicycle,
  faBus,
  faTrain,
  faSubway,
  faTram,
  faShip,
  faExclamationTriangle,
  faRoute,
  faSpinnerThird,
  faChevronRight,
  faCircle,
  faBusAlt,
  faArrowRight,
  faArrowLeft,
} from "@fortawesome/pro-duotone-svg-icons";
import { faMapMarkerAlt } from "@fortawesome/pro-solid-svg-icons";
import store from "../../redux/store";
import GlobalStyle from "../../GlobalStyle";
import Header from "../Header/Header";
import Footer from "../Footer/Footer";
import Main from "../Main/Main";
import { Container } from "./App.styled";

library.add(
  fab,
  faWalking,
  faBicycle,
  faBus,
  faBusAlt,
  faTrain,
  faSubway,
  faTram,
  faShip,
  faMapMarkerAlt,
  faExclamationTriangle,
  faRoute,
  faSpinnerThird,
  faChevronRight,
  faCircle,
  faArrowRight,
  faArrowLeft,
);

const App = () => (
  <Provider store={store}>
    <GlobalStyle />
    <Container>
      <BrowserRouter>
        <Header />
        <Main />
        <Footer />
      </BrowserRouter>
    </Container>
  </Provider>
);

export default App;
