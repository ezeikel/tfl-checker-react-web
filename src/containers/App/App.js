import React from 'react';

import Main from '../Main/Main';

const App = () => (
  <div>
    <Main />
  </div>
)

export default App;
