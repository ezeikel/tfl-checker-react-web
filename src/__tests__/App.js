import React from 'react';
import { render } from '@testing-library/react';
import App from '../components/App';

test('renders without crashing', () => {
  const { container } = render(
    <App />
  );
  expect(container).toMatchSnapshot();
});