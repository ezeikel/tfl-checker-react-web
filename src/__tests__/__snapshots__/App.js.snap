// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    class="sc-fzqMdD fuBuAJ"
  >
    <header
      class="sc-AxiKw haLAnR"
    >
      <section
        class="sc-AxhCb fmZXwX"
      >
        <a
          href="/"
        >
          TfL Checker
        </a>
      </section>
      <nav
        class="sc-AxjAm iMxqKn"
      >
        <ul
          class="sc-AxirZ cBkftF"
        >
          <li>
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Status updates
            </a>
          </li>
          <li>
            <a
              href="/trip-planner"
            >
              Plan a journey
            </a>
          </li>
        </ul>
      </nav>
    </header>
    <main
      class="sc-fzqAui kdCRTf"
    >
      <div
        class="sc-fznyAO dEIstO"
      />
    </main>
    <footer
      class="sc-AxhUy cyseKh"
    >
      <section
        class="sc-AxgMl bWmTSR"
      >
        <a
          href="/"
        >
          TfL Checker
        </a>
      </section>
      <section
        class="sc-AxheI ghZMUJ"
      >
        <div
          class="sc-Axmtr gooHpQ"
        >
          <span>
            This website was built using
          </span>
          <span>
            React
          </span>
        </div>
        <div
          class="sc-AxmLO iZUaIX"
        >
          <span>
            I also made versions using
          </span>
          <ul>
            <li>
              Vanilla JavaScript (no framework)
            </li>
            <li>
              Angular
            </li>
            <li>
              Vue
            </li>
            <li>
              Svelte
            </li>
            <li>
              AngularJS
            </li>
            <li>
              BackboneJS
            </li>
          </ul>
        </div>
      </section>
      <section
        class="sc-fzozJi kuavcF"
      >
        <div>
          © TfL Checker. All rights reserved.
        </div>
        <div>
          Made with 
          <span>
            ♡
          </span>
           in 
          <span>
            South London
          </span>
        </div>
      </section>
    </footer>
  </div>
</div>
`;
